Library/Homebrew/formulary.rb

```diff
@@ -942,8 +943,7 @@ def self.tap_loader_for(tapped_name, warn:)
 
     if tap.core_tap? && !Homebrew::EnvConfig.no_install_from_api?
       if type == :alias
-        alias_name = tapped_name[HOMEBREW_TAP_FORMULA_REGEX, 3]
-        return AliasAPILoader.new(alias_name)
+        return AliasAPILoader.new(name)
```

Comment:
Could you add this line back in and remove the regex usage from a few lines above? It seems like the type checker has failed because name can be nil as a result of this change on line 952 when calling `find_formula_in_tap`.

This was showing up when running: `brew audit --skip-style --except=installed,version --tap=homebrew/core --debug`

Reply:
With this patch, the audit started working again for me locally.

```diff
diff --git a/Library/Homebrew/formulary.rb b/Library/Homebrew/formulary.rb
index 28534bfa0..a20eb3457 100644
--- a/Library/Homebrew/formulary.rb
+++ b/Library/Homebrew/formulary.rb
@@ -909,7 +909,7 @@ module Formulary
     alias_name = tap.core_tap? ? name : "#{tap}/#{name}"
 
     if (possible_alias = tap.alias_table[alias_name].presence)
-      name = possible_alias[HOMEBREW_TAP_FORMULA_REGEX, 3]
+      name = possible_alias
       type = :alias
     elsif (new_name = tap.formula_renames[name].presence)
       old_name = name
@@ -943,7 +943,8 @@ module Formulary
 
     if tap.core_tap? && !Homebrew::EnvConfig.no_install_from_api?
       if type == :alias
-        return AliasAPILoader.new(name)
+        alias_name = tapped_name[HOMEBREW_TAP_FORMULA_REGEX, 3]
+        return AliasAPILoader.new(alias_name)
       elsif Homebrew::API::Formula.all_formulae.key?(name)
         return FormulaAPILoader.new(name)
       end
```

Reply:
Done, but then I had to apply a fix in line 953, to make it work.